internal/node/hybrid/kubelet_cert_validator.go

```diff
@@ -0,0 +1,65 @@
+package hybrid
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/eks-hybrid/internal/kubelet"
+)
+
+// ValidateKubeletCert checks if there is an existing kubelet certificate and validates it against the cluster's CA
+func (hnp *HybridNodeProvider) ValidateKubeletCert() error {
```

Comment:
not a blocker: I would implement this separate from the hnp, afaik there is no reason why it needs to be tied to it. I know we will want to use this from the `debug` command as well